go/kbfs/libkbfs/conflict_resolver.go

```diff
@@ -3324,6 +3325,57 @@ func (cr *ConflictResolver) recordFinishResolve(
 	err = serializeAndPutConflicts(cr.config, db, key, conflictsSoFar)
 }
 
+func (cr *ConflictResolver) makeDirtyBcache(
+	ctx context.Context, kmd KeyMetadata) (
+	dirtyBcache DirtyBlockCacheSimple, cleanupFn func(context.Context),
+	err error) {
+	var dbc *DiskBlockCacheLocal
+	if cr.config.IsTestMode() {
+		// Enable the disk limiter if one doesn't exist yet.
+		_ = cr.config.(*ConfigLocal).EnableDiskLimiter(os.TempDir())
+
+		dbc, err = newDiskBlockCacheLocalForTest(
+			cr.config, syncCacheLimitTrackerType)
+		if err != nil {
+			return nil, nil, err
+		}
+		cleanupFn = dbc.Shutdown
+	} else {
+		tempDir, err := ioutil.TempDir(cr.config.StorageRoot(), "kbfscr")
+		if err != nil {
+			return nil, nil, err
+		}
+		dirCleanupFn := func(_ context.Context) {
+			err := os.RemoveAll(tempDir)
+			if err != nil {
+				cr.log.CDebugf(ctx, "Error cleaning up tempdir %s: %+v",
+					tempDir, err)
+			}
+		}
+		dbc, err = newDiskBlockCacheLocal(
+			cr.config, crDirtyBlockCacheLimitTrackerType, tempDir)
+		if err != nil {
+			dirCleanupFn(ctx)
+			return nil, nil, err
+		}
+		cleanupFn = func(ctx context.Context) {
+			dbc.Shutdown(ctx)
+			dirCleanupFn(ctx)
+		}
+	}
+
+	err = dbc.WaitUntilStarted()
+	if err != nil {
+		if cleanupFn != nil {
```

Comment:
Maybe I'm just failing at following control flow, but it doesn't seem to me like `cleanupFn` can be nil.

Reply:
Never mind, I see now I think.

Reply:
Actually I think it can't be `nil` (it could be in an earlier iteration of the code), but it doesn't seem bad to continue to do the check just in case.